code/modules/power/rtg.dm

```diff
@@ -0,0 +1,93 @@
+// Radioisotope Thermoelectric Generator (RTG)
+// Simple power generator that would replace "magic SMES" on various derelicts.
+
+/obj/machinery/power/rtg
+	name = "radioisotope thermoelectric generator"
+	desc = "A simple nuclear power generator, used in small outposts to reliably provide power for decades."
+	icon = 'icons/obj/power.dmi'
+	icon_state = "rtg"
+	density = 1
+	anchored = 1
+	use_power = 0
+	var/power_gen = 1000 // Enough to power a single APC. Up to 4000 output with T4 capacitor.
+	var/board_path = /obj/item/weapon/circuitboard/machine/rtg
+
+/obj/item/weapon/circuitboard/machine/rtg
+	name = "circuit board (RTG)"
+	build_path = /obj/machinery/power/rtg
+	origin_tech = "programming=2;materials=4;powerstorage=3;engineering=2"
+	req_components = list(
+		/obj/item/stack/cable_coil = 5,
+		/obj/item/weapon/stock_parts/capacitor = 1,
+		/obj/item/stack/sheet/mineral/uranium = 10) // We have no Pu-238, and this is the closest thing to it.
+	//def_components = list(/obj/item/weapon/stock_parts/cell = /obj/item/weapon/stock_parts/cell/high/empty)
+
+/obj/machinery/power/rtg/New()
+	..()
+	var/obj/item/weapon/circuitboard/machine/B = new board_path(null)
+	B.apply_default_parts(src)
+	connect_to_network()
+
+/obj/machinery/power/rtg/initialize()
+	..()
+	connect_to_network()
+
+/obj/machinery/power/rtg/process()
+	..()
+	add_avail(power_gen)
+
+/obj/machinery/power/rtg/RefreshParts()
+	power_gen = initial(power_gen)
+	var/part_level = 0
+	for(var/obj/item/weapon/stock_parts/SP in component_parts)
+		part_level += SP.rating
+
+	power_gen *= part_level
```

Comment:
```power_gen = initial(power_gen)*part_level```, and then remove the first set to initial.
no point accessing/setting the var twice.